xyz-hub-service/src/main/java/com/here/xyz/hub/task/FeatureTaskHandler.java

```diff
@@ -411,40 +435,89 @@ private static XyzResponse transform(byte[] value) throws JsonProcessingExceptio
   }
 
   static <T extends FeatureTask> void setAdditionalEventProps(T task, Connector connector, Event event) {
-    event.setMetadata(task.getJwt().metadata);
+    setAdditionalEventProps(new NotificationContext(task, false), connector, event);
+  }
+
+  static void setAdditionalEventProps(NotificationContext nc, Connector connector, Event event) {
+    event.setMetadata(nc.jwt.metadata);
     if (connector.trusted) {
-      event.setTid(task.getJwt().tid);
-      event.setAid(task.getJwt().aid);
-      event.setJwt(task.getJwt().jwt);
+      event.setTid(nc.jwt.tid);
+      event.setAid(nc.jwt.aid);
+      event.setJwt(nc.jwt.jwt);
+    }
+  }
+
+  static void setLatestSeenContentVersion(Space space, int version) {
+    if (space.isEnableHistory() && version > 0)
+      latestSeenContentVersions.computeIfPresent(space.getId(), (spaceId, currentVersion) -> Math.max(currentVersion, version));
+  }
+
+  /**
+   * Schedules the sending of a {@link ContentModifiedNotification} to the modification SNS topic and all listeners registered for it.
+   * If some notification was already scheduled in the last time interval (specified by {@link #CONTENT_MODIFICATION_INTERVAL}), no further
+   * notification will be scheduled for the current task.
+   * @param task The {@link FeatureTask} which triggers the notification
+   * @param <T>
+   */
+  private static <T extends FeatureTask> void scheduleContentModifiedNotification(T task) {
+    NotificationContext nc = new NotificationContext(task, false);
+    scheduleContentModificationNotificationIfAbsent(nc, contentModificationTimers, CONTENT_MODIFICATION_INTERVAL, false);
+    scheduleContentModificationNotificationIfAbsent(nc, contentModificationAdminTimers, CONTENT_MODIFICATION_ADMIN_INTERVAL, true);
+  }
+
+  private static void scheduleContentModificationNotificationIfAbsent(NotificationContext nc, ConcurrentHashMap<String, Long> timerMap,
+      long interval, boolean adminNotification) {
+    if (!timerMap.containsKey(nc.space.getId())) {
+      //Schedule a new notification
+      long timerId = Service.vertx.setTimer(interval, tId -> {
```

Comment:
Despite of the fact of multiple timers in case execution reaches this point at the same time for many threads, it should not harm, since only one will remain active by executing "Service.vertx.cancelTimer(timerId);"